### PR TITLE
Cleanup the OFI detection m4

### PR DESCRIPTION
**Pull request by rhc54**
If OFI isn't installed, don't abort unless the user specifically asked for it.

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by rhc54** [comment]
@ggouaillardet I ran into a problem when trying to build on my Mac where there is no libfabric installed. I had to add a little protection to ensure we handled that case. Let me know what you think.

File: include/pmix_common.h.in [hunk 1]
```diff
@@ -1478,6 +1478,9 @@ static inline void* pmix_calloc(size_t n, size_t m)
 #define PMIX_CHECK_RANK(a, b) \
     ((a) == (b) || (PMIX_RANK_WILDCARD == (a) || PMIX_RANK_WILDCARD == (b)))
 
+#define PMIX_NSPACE_INVALID(a) \
```
Comment (bwbarrett) [inline include/pmix_common.h.in:1481]: unrelated? :)

**Review by bwbarrett — APPROVED** [review]

File: include/pmix_common.h.in (hunk 1 above)
Comment (rhc54) [reply include/pmix_common.h.in:1481]: Yeah it is - just happened to be in there at the same time, and not worth a separate pass thru the CI 😄
